### PR TITLE
null checks

### PR DESCRIPTION
**Pull request by likewei92**
These checks are necessary now that we allow switching roots

### CODE REVIEW

File: src/focus.service.ts [hunk 1]
```diff
@@ -281,7 +281,7 @@ export class FocusService {
       return;
     }
 
-    for (let el = this.selected; el !== root; el = el.parentElement) {
+    for (let el = this.selected; !!el && el !== root; el = el.parentElement) {
```
Comment (SimonSchick) [inline src/focus.service.ts]: Not sure if you need the negation here as null evaluates to false.

**Comment by connor4312** [comment]
It should never be possible that an element outside of the root is ever focused, these checks should not be needed.

**Comment by likewei92** [comment]
@connor4312 it is possible if we switch the root?

**Comment by connor4312** [comment]
The goal of adding [setDefaultFocus](https://github.com/WatchBeam/arcade-machine/blob/bb1c7d22fe8aebb2bbe9b535e59ebc025ccb8817/src/focus.service.ts#L286) was to make sure that you never left the root. Doing so puts you in an inconsistent state and leads to lots of weird bugs, some of which this PR patches, but it would be better not to enter that state at all.

**Comment by likewei92** [comment]
Closing this PR because we're changing everything in frontend2 to be within the xbox-wrapper, and in that case we have everything behaving nicely inside the root

**Pull request closed without merging**